test/e2e/addon_update.go

```diff
@@ -210,13 +210,25 @@ var _ = KubeDescribe("Addon update", func() {
 		// Reduce the addon update intervals so that we have faster response
 		// to changes in the addon directory.
 		// do not use "service" command because it clears the environment variables
-		sshExecAndVerify(sshClient, "sudo TEST_ADDON_CHECK_INTERVAL_SEC=1 /etc/init.d/kube-addons restart")
+		if testContext.OSDistro == "debian" {
+			sshExecAndVerify(sshClient, "sudo TEST_ADDON_CHECK_INTERVAL_SEC=1 /etc/init.d/kube-addons restart")
+		} else if testContext.OSDistro == "trusty" {
+			sshExecAndVerify(sshClient, "sudo initctl restart kube-addons TEST_ADDON_CHECK_INTERVAL_SEC=1")
+		} else {
+			Failf("Unsupported OS distro type %s", testContext.OSDistro)
```

Comment:
AFAIK, this test case fails on CoreOS which uses systemd service https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/coreos/master.yaml#l180. I am not clear about their status or plan for make e2e all green on CoreOS. Let's cc @yifan-gu to notify them about this change.